### DISCUSSION TITLE
Verdaccio Deployment to EKS Clusters

### DISCUSSION THREAD

**Discussion by HackerM0nk** [verdaccio/charts#127] [Q&A]
Hi guys,

Could someone please explain the high-level architecture of their Verdaccio deployments on Kubernetes? I'm having trouble setting it up on EKS Cluster using the Helm chart and haven't been able to resolve them using the existing documentation available on the official docs and other resources on the internet / mentioned in other discord channels / issues here.

The following are the issues/confusions I'm experiencing:

- [1] How many pods should be running in the ReplicaSet for an EKS cluster **HA setup** of Verdaccio?
- [2] Because EBS-based volumes do not support the "ReadWriteMany" property, any new pods that start up after a restart get errors such as the **"FailedAttachVolume Multi-Attach"** error for the volume.
- [3] Several issues with installing Prometheus monitoring plugins. When I try to install those packages, I get a "package not found" error, resulting in a **crashloopbackoff**.
- [4] How to prevent **unscoped internal packages** from dependency confusion attacks? The documentation only mentions it's only possible to prevent the issue with scoped packages. Can we somehow configure this at Verdaccio level itself?

 I would be extremely grateful, If anyone could kindly offer guidance or assistance with these challenges.

**Comment by juanpicado** [comment]
Cc @verdaccio/kubernetes fyi
